### ISSUE TITLE
[Bug] [iOS] Rendering Line shape crashes the app when building using iOS SDK 15.2

### ISSUE THREAD

**Issue by activa** [xamarin/Xamarin.Forms#14986]
### Description

When using iOS SDK 15.2 (with Xcode 13.2), any Xamarin Forms app that uses Line shapes crashes immediately when the line is rendered.

### Steps to Reproduce

1. Add a Line shape in XAML (or in code)
2. Compile using iOS SDK 15.2
3. Run app

For example, this page will crash the app:
```
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="XFLineBug.MainPage">

    <Grid>
        <Line X1="10" Y1="10" X2="100" Y2="50" Stroke="Red" StrokeThickness="3"/>
    </Grid>

</ContentPage>
```

Exception thrown: `Could not initialize an instance of the type 'CoreGraphics.CGPath': handle is null.`

Repro project: [XFLineBug.zip](https://github.com/xamarin/Xamarin.Forms/files/7731122/XFLineBug.zip)
(this is a clean new Xamarin Forms iOS project with the MainPage changed to show a line shape)

### Expected Behavior

Line is shown on screen

### Actual Behavior

App crashes immediately.

100% reproducible

### Basic Information

- Version with issue: 5.0.0.2125 and 5.0.0.2291 (possibly earlier versions as well)
- Last known good version: unknown
- Platform Target Frameworks: 
  - iOS:  15.2 (Xamarin.iOS 15.4.0)
  - Android: N/A
  - UWP:  N/A
- Android Support Library / AndroidX Version: N/A
- Affected Devices: all iOS devices

### Environment

Visual Studio 2022 on Windows
Visual Studio 2019 with Xamarin.iOS 15.4.0, Xcode 13.2 and iOS SDK 15.2 installed on Mac agent

**Comment by DoubleDBE** [comment]
Can confirm, having the same issue. Same environment as OP.

**Comment by NiknakSi** [comment]
Seeing this issue as well using Visual Studio for Mac on macOS. VS 8.10.16 (build 2), Xamarin.iOS 15.4.0.0, Xcode 13.2 (13C90), XF 5.0.0.2125 or 5.0.0.2291.

**Comment by adamzucchi** [comment]
Can also confirm.

**Comment by robintschroeder** [comment]
Confirmed. Please fix!

**Comment by activa** [comment]
@jsuarezruiz Can this be looked into ASAP? It's a crippling bug that needs to be addressed immediately. Anyone using Line shapes in an iOS app is no longer able to build and run the app - at all! 
There's no workaround other than going back to Xcode 13.1 or earlier (if that's even possible)

**Comment by activa** [comment]
Stack trace:
```
Exception : Could not initialize an instance of the type 'CoreGraphics.CGPath': handle is null.
It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false.

at ObjCRuntime.DisposableObject.InitializeHandle (System.IntPtr handle, System.Boolean verify) [0x0003e] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/ObjCRuntime/DisposableObject.cs:85
at ObjCRuntime.DisposableObject..ctor (System.IntPtr handle, System.Boolean owns, System.Boolean verify) [0x00006] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/ObjCRuntime/DisposableObject.cs:51
at CoreFoundation.NativeObject..ctor (System.IntPtr handle, System.Boolean owns, System.Boolean verify) [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/CoreFoundation/NativeObject.cs:46
at CoreFoundation.NativeObject..ctor (System.IntPtr handle, System.Boolean owns) [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/CoreFoundation/NativeObject.cs:41
at CoreGraphics.CGPath..ctor (System.IntPtr handle, System.Boolean owns) [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/CoreGraphics/CGPath.cs:109
at CoreGraphics.CGPath.MakeMutable (System.IntPtr source, System.Boolean owns) [0x0000f] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/CoreGraphics/CGPath.cs:518
at CoreGraphics.CGPath.CopyByTransformingPath (CoreGraphics.CGAffineTransform transform) [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/CoreGraphics/CGPath.cs:573
at Xamarin.Forms.Platform.iOS.ShapeLayer.BuildRenderPath () [0x003de] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Shapes\ShapeRenderer.cs:449
at Xamarin.Forms.Platform.iOS.ShapeLayer.UpdatePathStrokeBounds () [0x0004a] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Shapes\ShapeRenderer.cs:584
at Xamarin.Forms.Platform.iOS.ShapeLayer.UpdateShape (CoreGraphics.CGPath path) [0x00034] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Shapes\ShapeRenderer.cs:289
at Xamarin.Forms.Platform.iOS.LineView.UpdateShape () [0x0002a] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Shapes\LineRenderer.cs:106
at Xamarin.Forms.Platform.iOS.LineView.UpdateX1 (System.Double x1) [0x0000c] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Shapes\LineRenderer.cs:80
at Xamarin.Forms.Platform.iOS.LineRenderer.UpdateX1 () [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Shapes\LineRenderer.cs:54
at Xamarin.Forms.Platform.iOS.LineRenderer.OnElementChanged (Xamarin.Forms.Platform.iOS.ElementChangedEventArgs`1[TElement] args) [0x0002a] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Shapes\LineRenderer.cs:31
at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x00172] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementRenderer.cs:296
at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementRenderer.cs:158
at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00032] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:240
```

**Comment by cfzdevelop** [comment]
Having similar error. I have a LaunchStoryboard as splash screen. The splash screen will be shown. But the app doesn‘t change to main window. This is happening since update to latest XF (*.2291) and XCode 15.2.

**Comment by matt-uib** [comment]
We see the same error here since we updated to Xcode 13.2 and Visual Studio for Mac 8.10.

**Comment by jfversluis** [comment]
Hey all, a PR for this is open now, would you be able to grab the NuGet [as described here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) and let us know if this fixes this issue? That will greatly speed up the review process. I'm not sure this is the right or final fix, but this might you get unblocked for the time being while we get past the holidays and New Years as nothing many people are out-of-office right now and things might not move very fast right now.

Besides verifying if this particular issue is fixed also be sure to check other scenarios in the same area to make sure that this fix doesn't accidentally has side-effects 🙂 

Thanks!

**Comment by NiknakSi** [comment]
Thanks @jfversluis - updated XF to 5.0.0.7924 and the crash with Line has been resolved and the Lines render as they did before. A very limited test but looks promising. Thanks for taking a look for us.

**Comment by adamzucchi** [comment]
> Hey all, a PR for this is open now, would you be able to grab the NuGet [as described here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) and let us know if this fixes this issue? That will greatly speed up the review process. I'm not sure this is the right or final fix, but this might you get unblocked for the time being while we get past the holidays and New Years as nothing many people are out-of-office right now and things might not move very fast right now.
> 
> Besides verifying if this particular issue is fixed also be sure to check other scenarios in the same area to make sure that this fix doesn't accidentally has side-effects 🙂
> 
> Thanks!

I also tested the 5.0.0.7924 preview for the fix above and am no longer seeing the crash, thanks!

**Comment by cfzdevelop** [comment]
> > Hey all, a PR for this is open now, would you be able to grab the NuGet [as described here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) and let us know if this fixes this issue? That will greatly speed up the review process. I'm not sure this is the right or final fix, but this might you get unblocked for the time being while we get past the holidays and New Years as nothing many people are out-of-office right now and things might not move very fast right now.
> > Besides verifying if this particular issue is fixed also be sure to check other scenarios in the same area to make sure that this fix doesn't accidentally has side-effects 🙂
> > Thanks!
> 
> I also tested the 5.0.0.7924 preview for the fix above and am no longer seeing the crash, thanks!

How did you retrieve this nuget version? Pre-release channel doesn‘t list it. I‘ve tried setting version in csproj manually and then restoring. But without success. Output telling me the version 5.0.0.7924 not found.

**Comment by adamzucchi** [comment]
> > > Hey all, a PR for this is open now, would you be able to grab the NuGet [as described here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) and let us know if this fixes this issue? That will greatly speed up the review process. I'm not sure this is the right or final fix, but this might you get unblocked for the time being while we get past the holidays and New Years as nothing many people are out-of-office right now and things might not move very fast right now.
> > > Besides verifying if this particular issue is fixed also be sure to check other scenarios in the same area to make sure that this fix doesn't accidentally has side-effects 🙂
> > > Thanks!
> > 
> > 
> > I also tested the 5.0.0.7924 preview for the fix above and am no longer seeing the crash, thanks!
> 
> How did you retrieve this nuget version? Pre-release channel doesn‘t list it. I‘ve tried setting version in csproj manually and then restoring. But without success. Output telling me the version 5.0.0.7924 not found.

In the message from @jfversluis a link was included which references the following Nuget Feed URL: 
https://aka.ms/forms-prs/index.json

In Visual Studio you will want to setup a new Nuget Feed and then in your solution update the Xamarin.Forms Nuget to 5.0.0.7924 when the new Nuget Feed you just setup is chosen as the source feed. Good luck!

**Comment by cfzdevelop** [comment]
> > > > Hey all, a PR for this is open now, would you be able to grab the NuGet [as described here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) and let us know if this fixes this issue? That will greatly speed up the review process. I'm not sure this is the right or final fix, but this might you get unblocked for the time being while we get past the holidays and New Years as nothing many people are out-of-office right now and things might not move very fast right now.
> > > > Besides verifying if this particular issue is fixed also be sure to check other scenarios in the same area to make sure that this fix doesn't accidentally has side-effects 🙂
> > > > Thanks!
> > > 
> > > 
> > > I also tested the 5.0.0.7924 preview for the fix above and am no longer seeing the crash, thanks!
> > 
> > 
> > How did you retrieve this nuget version? Pre-release channel doesn‘t list it. I‘ve tried setting version in csproj manually and then restoring. But without success. Output telling me the version 5.0.0.7924 not found.
> 
> In the message from @jfversluis a link was included which references the following Nuget Feed URL: https://aka.ms/forms-prs/index.json
> 
> In Visual Studio you will want to setup a new Nuget Feed and then in your solution update the Xamarin.Forms Nuget to 5.0.0.7924 when the new Nuget Feed you just setup is chosen as the source feed. Good luck!

Thank you for the instructions. Downloaded 5.0.0.7924 and my app is working again. Hope that there is no side effect.

**Comment by chaoyebugao** [comment]
TMD...

**Comment by chaoyebugao** [comment]
Well, can you guys fix this officially? It's a serious problem...

**Comment by jfversluis** [comment]
@chaoyebugao I understand, however a lot of people are off right now because they are celebrating Christmas and it's almost a new year on their calendar. A new update won't be out until halfway in January, so there is no need to rush at this time. I understand it's a problem and I'm sorry about that, but there is not much I can do right now.

Although we can't officially recommend and support it, the NuGet I referenced to might be a temporary solution. If you choose to take that route I would be very thankful if you could let me know the results, that will speed up merging this as a official fix. Thanks!

**Comment by chrisstaley** [comment]
@jfversluis I can confirm that the PR fixes the issue in our app.

And thanks for resolving this in a quick, yet responsible manner. I appreciate that.

**Comment by activa** [comment]
Why does this bug still have the tag s/unverified ?

**Comment by jfversluis** [comment]
@activa don't worry about the tags too much :)

@sady4850 will look at macOS as well!

**Comment by jfversluis** [comment]
So, while looking into this I can't reproduce anymore with the very latest version of Xcode (Version 13.2.1 (13C100)) and Xamarin.iOS (15.2.0.17) for both iOS and macOS. Would you be able to try it with all the latest versions?

**Comment by activa** [comment]
I installed XCode 13.2.1 and tried again but the issue remains. I do have Xamarin.iOS 15.4.0.0 though (which I think is the latest stable)

**Comment by sady4850** [comment]
I have latest xCode, Xamarin.iOS, Xamarin.MacOS and it crashes in both macOS and iOS.
```
Apple Developer Tools
Xcode 13.2.1 (19586)
Build 13C100

Xamarin.Mac
Version: 8.4.0.0 (Visual Studio Community)
Hash: 8fc41ae82
Branch: xcode13.2
Build date: 2021-12-09 01:07:56-0500

Xamarin.iOS
Version: 15.4.0.0 (Visual Studio Community)
Hash: 8fc41ae82
Branch: xcode13.2
Build date: 2021-12-09 01:07:56-0500
```

**Comment by jfversluis** [comment]
Interesting. @dalexsoto or maybe even @rachelkang does this sound familiar to you from recent history?

I definitely saw this crashing last month, but when I try to reproduce now (without the fix on our side) I'm not able to.

**Comment by dalexsoto** [comment]
@jfversluis that doesn’t sound familiar, that said Xcode 13.2 had a lot of native API breaking changes which means that they were doing some cleanup so probably this is hitting something they “fixed”. Now about you not being able to repro, I would just triple check which version of Xcode you have selected in both xcode-select and VS so they match on 13.2 🙂

**Comment by activa** [comment]
@jfversluis It may not be related to the Xcode version at all. A colleague of mine saw this bug after upgrading Visual Studio on the Mac to the latest version. Maybe Xamarin.iOS version 15.4.0.0 is the culprit

**Comment by jfversluis** [comment]
@dalexsoto yeah in line with what @activa is saying; the people reporting this and me seem to use the same Xcode version, the only difference is the Xamarin.iOS version at this point. When is a new Xamarin.iOS (public) update planned?Or maybe in the meantime I should try and see if I can downgrade mine to see if I suddenly start seeing it again

**Comment by sady4850** [comment]
@jfversluis Did you try to repro it with Xamarin.iOS version 15.4.0.0? I'm asking because previously you said that you have 15.2.0.17 version

**Comment by activa** [comment]
My colleague's setup is Xcode 13.1 (iOS SDK 15.0) with Xamarin.iOS 15.4.0.0. The bug is 100% reproducible with that setup. 
My setup is the same except I'm using Xcode 13.2 (iOS SDK 15.2). Bug is also reproducible. When I upgraded to Xcode 13.2 I also updated Xamarin at the same time so that could be why we initially thought it was caused by Xcode 13.2 instead of Xamarin.iOS 15.4.0.0

**Comment by EeeDotZee** [comment]
Just to confirm I'm having the exact same issue, but only when i try to deploy my app to a simulator that didn't have the app installed before. 
I got everything up to date (xcode, VS, xamarin libraries, xamarin forms, etc.)

**Comment by m1dst** [comment]
Tested against XF 5.0.0.7954 and confirm it still doesn't work.  I am using XC '13.2.1' and Xamarin.iOS '15.2.0.17'

**Comment by jsuarezruiz** [comment]
The fix https://github.com/xamarin/Xamarin.Forms/pull/14993 will be included in the next SR.

**Comment by jfversluis** [comment]
@m1dst SR9 will be released next week (if all goes well) and since the PR to fix this issue is now merged that SR should have the fix for this issue but is not released.

To verify that my fix is working correctly I am asking people to test a specific NuGet that results from a CI build. Those versions should only be used for testing that specific fix. I see you mention 7954, while the version that was attached to this fix is 7924. So you have probably been testing the wrong version :)

**Comment by m1dst** [comment]
@jfversluis Thanks for confirming.  I didn't realise the issue was pinned to that version.  Switching back a few versions to 7924 things start working again.  Nice one!

**Comment by MosCD3** [comment]
This issue shouldn't be closed until the official fix is shipped and released.

**Comment by jfversluis** [comment]
@MosCD3 our process works a bit different. The official fix has been merged and should be released soon.

**Comment by activa** [comment]
> @MosCD3 our process works a bit different. The official fix has been merged and should be released soon.

It's not different. I think the vast majority of software companies handle bugs the same way. Bugs are closed when they're resolved and merged into the release branch. At least that's how we do it.

**Comment by slprog1** [comment]
Hello, guys!
I was wondering if anybody could tell when the fix will be available? I see the planning date is Jan 19th, but still it's not clear when it will be released

**Comment by jfversluis** [comment]
@slprog1 the release is going through some checks right now and unless something weird is found there it should be this week.

**Comment by slprog1** [comment]
Hello, @jfversluis! Thanks for clarifying! What confused me is the release date I saw under the "labels" section 
Anyway, looking forward for the fix - we can't publish our new version of the app with critical bug fixes, because we have some graphics on the screens, thus the app simply crashes ((

**Comment by jfversluis** [comment]
I can imagine. The way I'm going about this now is working with a target date. By that date I want to have it published, but it can be a few days late (or early). There is some quality checks and tests, etc. we have to run before releasing. If anything comes up there we want to of course fix that first :) I don't think there is much wrong with this release so should be soon 🤞

**Comment by slprog1** [comment]
Hello, @jfversluis!
Really sorry for bothering you, but is there any updates on the release date? Maybe the update is in the preview channel, so I could try to use it?

**Comment by jfversluis** [comment]
Still going through checks. If you really can't wait have a look here: https://github.com/xamarin/Xamarin.Forms/releases/tag/release-5.0.0-sr9

And download the nuget manually. I don't recommend it though, it might not be the final version if we still find something. It will be released when it's released. As soon as the checks are ready we will upload them, so until then I don't know any more about a date. Sorry

**Comment by reggiebIV** [comment]
> Still going through checks. If you really can't wait have a look here: https://github.com/xamarin/Xamarin.Forms/releases/tag/release-5.0.0-sr9
> 
> And download the nuget manually. I don't recommend it though, it might not be the final version if we still find something. It will be released when it's released. As soon as the checks are ready we will upload them, so until then I don't know any more about a date. Sorry

I am running in to the same issue but with a polygon instead of a line shape. I am assuming the same fix would resolve my issue as well?

**Comment by jfversluis** [comment]
@reggiebIV I would assume so! You can test as per the instructions in my comment above :)

**Comment by EeeDotZee** [comment]
I can confirm on my end that installing the latest Service update from nuget has fixed the issue on my side.
Thank you guys for the effort.

**Comment by jfversluis** [comment]
Thanks for the confirmation @EeeDotZee !

Everyone else; in case you missed it: the NuGet is out officially now! :)

**Comment by dingxiaolong** [comment]
> @jfversluis that doesn’t sound familiar, that said Xcode 13.2 had a lot of native API breaking changes which means that they were doing some cleanup so probably this is hitting something they “fixed”. Now about you not being able to repro, I would just triple check which version of Xcode you have selected in both xcode-select and VS so they match on 13.2 🙂

Dear I still have this issue on Xamarin.ios when try to create CGPDFDocument like this: 
public CGPDFDocument PDFFromArray(byte[] array)
		{
			var dp = new CGDataProvider(array, 0, array.Length);
			return new CGPDFDocument(new CGDataProvider(dp.Handle));
		}


![image](https://user-images.githubusercontent.com/21192107/160735558-d204b63e-315c-4f54-86f0-5ebec5f4bc58.png)

Even use the latest Xcode 13.3 and latest Xamarin 5.0.406.
Could you help check please ,many thanks .
